pkg/scaffold/resources/ingress.go

```diff
@@ -24,61 +23,77 @@ func CreateOkctlIngress(app v1alpha1.Application) (networkingv1beta1.Ingress, er
 	ingress.Annotations["alb.ingress.kubernetes.io/actions.ssl-redirect"] =
 		`{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}`
 
-	redirectPath := networkingv1beta1.HTTPIngressPath{
-		Path: "/*",
-		Backend: networkingv1beta1.IngressBackend{
-			ServiceName: "ssl-redirect",
-			ServicePort: intstr.IntOrString{
-				Type:   intstr.String,
-				StrVal: "use-annotation",
+	// https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/tasks/ssl_redirect/
+	pathType := networkingv1.PathTypeImplementationSpecific
+
+	redirectPath := networkingv1.HTTPIngressPath{
+		Path:     "/*",
+		PathType: &pathType,
+		Backend: networkingv1.IngressBackend{
+			Service: &networkingv1.IngressServiceBackend{
+				Name: "ssl-redirect",
+				Port: networkingv1.ServiceBackendPort{
+					Name: "use-annotation",
+				},
 			},
 		},
 	}
 
-	ingress.Spec.Rules[0].HTTP.Paths = append([]networkingv1beta1.HTTPIngressPath{redirectPath}, ingress.Spec.Rules[0].HTTP.Paths...)
+	ingress.Spec.Rules[0].HTTP.Paths = append([]networkingv1.HTTPIngressPath{redirectPath}, ingress.Spec.Rules[0].HTTP.Paths...)
 
 	return ingress, nil
 }
 
-func generateDefaultIngress() networkingv1beta1.Ingress {
-	return networkingv1beta1.Ingress{
+func generateDefaultIngress() networkingv1.Ingress {
+	return networkingv1.Ingress{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Ingress",
-			APIVersion: "networking.k8s.io/v1beta1",
+			APIVersion: apiVersion(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			Labels:      map[string]string{},
 			Annotations: map[string]string{},
 		},
-		Spec: networkingv1beta1.IngressSpec{
-			Rules: make([]networkingv1beta1.IngressRule, 1),
+		Spec: networkingv1.IngressSpec{
+			Rules: make([]networkingv1.IngressRule, 1),
 		},
 	}
 }
 
-func createGenericIngress(app v1alpha1.Application) (networkingv1beta1.Ingress, error) {
+func apiVersion() string {
+	return fmt.Sprintf("%s/%s", networkingv1.SchemeGroupVersion.Group, networkingv1.SchemeGroupVersion.Version)
+}
+
+func createGenericIngress(app v1alpha1.Application) (networkingv1.Ingress, error) {
 	hostURL, err := app.URL()
 	if err != nil {
-		return networkingv1beta1.Ingress{}, fmt.Errorf("getting application URL: %w", err)
+		return networkingv1.Ingress{}, fmt.Errorf("getting application URL: %w", err)
 	}
 
 	ingress := generateDefaultIngress()
 	ingress.ObjectMeta.Namespace = app.Metadata.Namespace
 
 	ingress.ObjectMeta.Name = app.Metadata.Name
 
-	ingress.Spec.Rules = []networkingv1beta1.IngressRule{
+	// https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/tasks/ssl_redirect/
+	pathType := networkingv1.PathTypeImplementationSpecific
```

Comment:
Are you sure `ImplementationSpecific` path type suits non-https-redirect routes?

ref: https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types

I'd assume `Prefix` was the most natural choice

Reply:
To be honest, I am not sure, but I am going to test it. I first used Prefix, but then I saw the AWS specific Ingress example using PathTypeImplementationSpecific (see doc in line 78).

Although that example use

```
 alb.ingress.kubernetes.io/ssl-redirect
```

while we use

```
alb.ingress.kubernetes.io/actions.ssl-redirect
```

so I need to dig a bit more into this.

Also, the link you provided says

> Implementations can treat this as a separate pathType or treat it identically to Prefix or Exact path types.

which _can_ mean that the ALB ingress controller can choose to treat it like Prefix, so I was thinking the ALB ingress controller knows best. But yeah, I'll more into it.

Reply:
I'm sure implementation specific is correct for the ssl redirect route, I just assumed prefix was the natural one for when you didn't want to handle routing somewhere else. Super interested in the result of your research